hibike/hibike_message.py

```diff
@@ -437,10 +448,13 @@ def blocking_read_generator(serial_conn, stop_event=None):
                 packets_buffer = packets_buffer[new_packet:]
             # Otherwise, there might be more incoming bytes for the current packet,
             # so we do a blocking read and try again
-            else:
+            elif not stop_event.is_set():
```

Comment:
i feel like putting this check *everywhere* is masking some underlying code structure issue. if you're worried about synchronization, then its a great time to
1) think hard about exactly how the threads/ processes can interleave and interfere
2) design the best, easiest, cleanest way to solve it
3) learn about python multi-threading/processing synchronization primitives